### ISSUE TITLE
compass-runtime-agent component installation fails

### ISSUE THREAD

**Issue by pbochynski** [kyma-project/kyma#12954]
**Description**
Some jobs fail to install compass-runtime-agent component due to missing kyma-integration namespace.
Examples:
- https://storage.googleapis.com/kyma-prow-logs/logs/kyma-integration-k3d-compass-dev/1476479504095383552/build-log.txt
- https://storage.googleapis.com/kyma-prow-logs/logs/kyma-integration-k3d-compass-dev/1476404005964877824/build-log.txt
- https://storage.googleapis.com/kyma-prow-logs/pr-logs/pull/kyma-project_kyma/12937/pre-main-kyma-integration-k3d-central-app-connectivity-compass/1476461323368796160/build-log.txt
- https://storage.googleapis.com/kyma-prow-logs/pr-logs/pull/kyma-project_kyma/12937/pre-main-kyma-integration-k3d-central-app-connectivity-compass/1476166990728007680/build-log.txt

Please verify if it is still required to install resources in kyma-integration namespace (we should move all resources to kyma-system namespace). If it is not possible consider adding that namespace creation to the prerequisites. Verify why namespace is not created until component installation fails.

**Comment by franpog859** [comment]
Currently, it's still required to create Role in `kyma-integration` namespace as Compass Runtime Agent operates on resources there

Compass Runtime Agent depends on Application Connector installation in regards to the `kyma-integration` namespace creation. Application Connector needs to be installed (`kyma-integration` namespace needs to be created) to enable the Compass Runtime Agent installation. Until now it worked seamlessly. In all of the failures, the installation of Application Connector didn’t even start yet, while the Compass Runtime installation already failed, causing the installation global failure. I suspect that something could change in some timeouts of reconciliation

The first time the error occurred is 22.12. I didn't find anything related to timeouts in the Reconciler at this time. I consulted it with the @kyma-project/jellyfish team and got a suggestion that we should bump the Reconciler to the latest version in the `kyma-cli` repo - [PR](https://github.com/kyma-project/cli/pull/1129) - as it was bumped last time 8 days ago and it may be related

There is also a [draft issue](https://github.com/kyma-incubator/reconciler/issues/634) that could enable an installation of components with dependencies to other components

**Comment by franpog859** [comment]
[The bump](https://github.com/kyma-project/cli/pull/1129) is merged, let's see if it helps 🤞🤞

**Comment by Disper** [comment]
Unfortunately [it didn't help](https://status.build.kyma-project.io/view/gs/kyma-prow-logs/logs/kyma-integration-k3d-compass-dev/1476781497200414720).

**Comment by franpog859** [comment]
Hmm, the pipeline still uses the same version of Kyma CLI `Kyma CLI version: stable-9cbaaad`. I'm checking how to bump it also there. It should be `stable-595ea23`

**Comment by franpog859** [comment]
Ok, I tagged the CLI so pipelines will use the right version 🤞🤞

**Comment by koala7659** [comment]
As a quick fix to unblock release I propose adding application-connector and compass-runtime agent to reconciler’s preComponents list in this PR: https://github.com/kyma-incubator/reconciler/pull/641/files. I will also implement decoupling of both components in a separate PR.

**Comment by pbochynski** [comment]
Maybe add kyma-integration namespace resource to compass-runtime component. This way it will be created if it doesn't exist. As a result chart will be independent of application-connector chart.

**Comment by koala7659** [comment]
Good idea, @pbochynski. I will implement it as a separate PR.
